### DISCUSSION TITLE
[QUESTION] - Exporting dashboard / charts as PDF ? Or downloading charts as csv via API ?

### DISCUSSION THREAD

**Discussion by devaale** [apache/superset#22696] [Q&A / Help] [answered]
Hello,

Been going over around the documentation of superset but haven't found if it's possible to export dashboards or charts as PDF's or download charts as .csv files via API ? I want to send API request from my application and get .csv file of a chart as a result - is that possible ?

**Comment by rusackas** [comment]
For Question 1, there's currently no capability to export PDFs. You're not the first to hope for this, so we'd love to see a PR from anyone so inclined.

For question 2, yep, you can download CSV via the API. You just need to hit
`http://{superset_host:port}/api/v1/chart/{chart id}/data/?format=csv`

**Comment by mschaefer-gresham** [comment]
@rusackas  Is this still not possible? Coming here because I was trying to figure out the same thing regarding exporting a dashboard as PDF from the API. But the question is quite old by now.

**Comment by hqms** [comment]
is there any workaround to download as pdf via api?

**Comment by rusackas** [comment]
I don't _believe_ it's possible, since the browser (not a backend API) does the work of generating the screenshot. @fisjac or @kgabryje or @eschutho can correct me if I'm mistaken.

**Reply by rusackas** [reply]
Also, if one of those folks could mark this as the correct answer, if I'm correct, we can close the case on this discussion. Thanks!

**Comment by fisjac** [comment]
There is now the ability to generate pdfs of charts and dashboards. This leverages the api to cache a screenshot of the dashboard. I believe it is technically possible to generate a pdf via hitting api endpoints, but doing so would require additional information that may be difficult to provide from outside of superset. @geido can you comment?

Files to look at:
`superset-frontend/src/dashboard/components/menu/DownloadMenuItems/DownloadScreenshot.tsx`

**Comment by geido** [comment, marked as answer]
To answer the question about downloading dashboard screenshots via API, yes that's possible using the POST `/<pk>/cache_dashboard_screenshot/` endpoint.
